app/assets/src/components/ui/controls/SearchBox.jsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { Search } from "semantic-ui-react";
+import { escapeRegExp, debounce } from "lodash";
+
+const delayCheckMatch = 1000;
+const waitHandleSearchChange = 500;
+
+class SearchBox extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.source = this.props.source;
+
+    this.resetComponent = this.resetComponent.bind(this);
+    this.handleSearchChange = this.handleSearchChange.bind(this);
+    this.handleResultSelect = this.handleResultSelect.bind(this);
+  }
+
+  componentWillMount() {
+    this.resetComponent();
+  }
+
+  resetComponent() {
+    this.setState({ isLoading: false, results: [], value: "" });
```

Comment:
Just initialized the state in the constructor.

Reply:
ok done